runtime/recipe/recipe.js

```diff
@@ -139,6 +139,18 @@ class Recipe {
     }
     if (!this._isValid()) {
       console.log(this.toString());
+      if (this._hasDuplicateView())
+        console.log("Has Duplicate View");
+      if (!this._views.every(view => view._isValid()))
+        console.log("Has Invalid View");
+      if (!this._particles.every(particle => particle._isValid()))
+        console.log("Has Invalid Particle");
+      if (!this._slots.every(slot => slot._isValid()))
+        console.log("Has Invalid Slot");
+      if (!this.viewConnections.every(connection => connection._isValid()))
+        console.log("Has Invalid ViewConnection");
+      if (!this.slotConnections.every(connection => connection._isValid()))
+        console.log("Has Invalid SlotConnection");
```

Comment:
Thank you, this is really great!
Maybe worth adding the name of the view/particle/etc that was invalid to make it even better?